src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -395,20 +545,52 @@ internal void TakeSnapshot(bool outputDelta)
                     {
                         lock (this.histogramBuckets.LockObject)
                         {
-                            this.primaryValue.SnapshotAsLong = this.primaryValue.CurrentAsLong;
-                            this.secondaryValue.SnapshotAsDouble = this.secondaryValue.CurrentAsDouble;
+                            this.snapshotValue.AsLong = this.runningValue.AsLong;
+                            this.histogramBuckets.SnapshotSum = this.histogramBuckets.RunningSum;
                             if (outputDelta)
                             {
-                                this.primaryValue.CurrentAsLong = 0;
-                                this.secondaryValue.CurrentAsDouble = 0;
+                                this.runningValue.AsLong = 0;
+                                this.histogramBuckets.RunningSum = 0;
                             }
 
-                            for (int i = 0; i < this.histogramBuckets.BucketCounts.Length; i++)
+                            for (int i = 0; i < this.histogramBuckets.RunningBucketCounts.Length; i++)
                             {
-                                this.histogramBuckets.AggregatedBucketCounts[i] = this.histogramBuckets.BucketCounts[i];
+                                this.histogramBuckets.SnapshotBucketCounts[i] = this.histogramBuckets.RunningBucketCounts[i];
                                 if (outputDelta)
                                 {
-                                    this.histogramBuckets.BucketCounts[i] = 0;
+                                    this.histogramBuckets.RunningBucketCounts[i] = 0;
+                                }
```

Comment:
Would there be a performance benefit to doing the following?

```csharp
for (int i = 0; i < this.histogramBuckets.RunningBucketCounts.Length; i++)
{
    this.histogramBuckets.SnapshotBucketCounts[i] = this.histogramBuckets.RunningBucketCounts[i];
}

if (outputDelta)
{
    for (int i = 0; i < this.histogramBuckets.RunningBucketCounts.Length; i++)
    {
        this.histogramBuckets.RunningBucketCounts[i] = 0;
    }
}
```

What about using `Array.Copy` and `Array.Clear`?

Reply:
|          Method | size |      Mean |     Error |    StdDev |
|---------------- |----- |----------:|----------:|----------:|
| ManualArrayZero |    5 |  5.098 ns | 0.0694 ns | 0.0649 ns |
|      ArrayClear |    5 |  8.213 ns | 0.0422 ns | 0.0395 ns |
| ManualArrayZero |   10 |  8.610 ns | 0.0791 ns | 0.0701 ns |
|      ArrayClear |   10 | 11.941 ns | 0.1635 ns | 0.1530 ns |
| ManualArrayZero |   20 | 15.596 ns | 0.1303 ns | 0.1219 ns |
|      ArrayClear |   20 | 12.075 ns | 0.1288 ns | 0.1205 ns |
| ManualArrayZero |   50 | 41.655 ns | 0.3566 ns | 0.3335 ns |
|      ArrayClear |   50 | 18.182 ns | 0.3551 ns | 0.4861 ns |
| ManualArrayZero |  100 | 82.170 ns | 1.6516 ns | 1.5449 ns |
|      ArrayClear |  100 | 34.746 ns | 0.5619 ns | 0.5256 ns |

Keep existing version since the default size 10 is faster